src/microbe_stage/Microbe.cs

```diff
@@ -1427,22 +1427,39 @@ private void OnIGotAddedToColony()
             return;
 
         var oldRotation = Rotation;
-        var vectorToParent = GlobalTransform.origin - ColonyParent.GlobalTransform.origin;
-        ChangeNodeParent(ColonyParent);
 
-        var vectorToParentRotated = vectorToParent.Rotated(Vector3.Down, Rotation.y);
-        var vectorToMembrane = Membrane.GetExternalOrganelle(vectorToParentRotated.x, vectorToParentRotated.y);
+        // Gets the global rotation of the parent
+        var globalParentRotation = ColonyParent.GlobalTransform.basis.GetEuler();
+
+        // A vector from the parent to me
+        var vectorFromParent = GlobalTransform.origin - ColonyParent.GlobalTransform.origin;
+
+        // This vector gets as if I had no rotation.
+        // This is important, because GetExternalOrganelle only works with non-rotated microbes
+        var vectorToParentRotated = (-vectorFromParent).Rotated(Vector3.Down, Rotation.y);
 
-        vectorToParentRotated = (-vectorToParent).Rotated(Vector3.Down, ColonyParent.Rotation.y);
-        var parentVectorToItsMembrane =
-            ColonyParent.Membrane.GetExternalOrganelle(vectorToParentRotated.x, vectorToParentRotated.y);
+        // This vector gets rotated as if the parent had no rotation.
+        var vectorFromParentRotated =
+            vectorFromParent.Rotated(Vector3.Down, globalParentRotation.y);
 
-        var requiredDistance = vectorToMembrane.Length() + parentVectorToItsMembrane.Length();
+        // Calculates the vector from my center to my membrane towards the parent.
+        // This vector gets rotated back, because I've rotated it two calls above.
+        var correctedVectorToParent = Membrane.GetExternalOrganelle(vectorToParentRotated.x, vectorToParentRotated.z)
+            .Rotated(Vector3.Up, Rotation.y);
```

Comment:
So, here, the correction applied is that the original `vectorToParent` pointed towards the parent's center, but you wanted it to point an actual external organelle instead (picking the closest), right? 

Either way I think the nature of this correction should be specified somewhere in the comment above or/and in the variable names, perhaps using intermediate variables if necessary.

Reply:
The method for the external organelle actually just gives me a point on the membrane nearest to the provided point. I have renamed the method to avoid confusion.

Reply:
I found this method rather confusing as well. It is much more clearer now I must say. Thanks!